src/spdx_tools/spdx3/model/dataset/dataset.py

```diff
@@ -0,0 +1,56 @@
+#  SPDX-FileCopyrightText: 2023 spdx contributors
+#
+#  SPDX-License-Identifier: Apache-2.0
+from enum import Enum, auto
+from typing import Dict, Optional
+
+from spdx_tools.common.typing.dataclass_with_properties import dataclass_with_properties
+from spdx_tools.common.typing.type_checks import check_types_and_set_values
+
+
+class ConfidentialityLevelType(Enum):
+    RED = auto()
+    AMBER = auto()
+    GREEN = auto()
+    CLEAR = auto()
+
+
+class DatasetAvailabilityType(Enum):
+    DIRECT_DOWNLOAD = auto()
+    SCRAPING_SCRIPT = auto()
+    QUERY = auto()
+    CLICKTHROUGH = auto()
+    REGISTRATION = auto()
+
+
+@dataclass_with_properties
+class Dataset:
+    data_collection_process: Optional[str] = None
+    intended_use: Optional[str] = None
+    dataset_size: Optional[int] = None
+    dataset_noise: Optional[str] = None
+    data_preprocessing_steps: Optional[str] = None
+    sensors: Optional[Dict[str, str]] = None
```

Comment:
This property doesn't have a maxCount neither, so I would expect a list here.

Reply:
I have one question concerning this: the markdown file for `DictionaryEntry` states that the property `value` is optional, how would we handle this? I would think `List[Dict[str, Optional[str]]]` would be the correct type hint.

Reply:
The only important difference would be the need for unique keys. So I don't know if this makes any sense, but from the definition in the spdx-3-model we could have two sensors with the same key, don't we?

Reply:
I believe this is still debated whether this is actually wanted in the spdx-3-model... If yes, then you are right, we have a problem

Reply:
Removed the `Optional`